services/business/checkers/sidecars_checker.go

```diff
@@ -0,0 +1,60 @@
+package checkers
+
+import (
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/services/models"
+	"strings"
+)
+
+type SidecarsChecker struct {
+	Namespace      string
+	ServiceName    string
+	ServiceVersion string
+	K8sClient      kubernetes.IstioClientInterface
+}
+
+const sidecarsCheckerType = "sidecar"
+const SidecarContainerImage = "docker.io/istio/proxy"
+
+func (checker *SidecarsChecker) Check() *models.IstioValidations {
+	if checker.K8sClient == nil {
+		k8sClient, err := kubernetes.NewClient()
+		if err != nil {
+			log.Warningf("Cannot create Kubernetes client to check for sidecars presence in service %v.%v.", checker.Namespace, checker.ServiceName)
+			return &models.IstioValidations{}
+		}
+		checker.K8sClient = k8sClient
+	}
+
+	pods, err := checker.K8sClient.GetServicePods(checker.Namespace, checker.ServiceName, checker.ServiceVersion, "")
```

Comment:
When I thought about this solution, I didn't think about coupling connection with validation logic. Otherwise, you have one checker only for service pods, but not for every single pod. If you see the route rules checker, it can validate any kind of array of route rules independently where they are from.
So this logic would come either from `istio_validations` or an specific `*_validations` file. If you want to use same endpoint as `istio_validations`, then just fetch service pods from there and pass it to the checker.